clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAO.scala

```diff
@@ -54,6 +58,33 @@ class HttpElasticsearchDAO private[dataaccess] (
     searchResponse flatMap foldScroll(Seq.empty[D], searchDefinition.keepAlive)
   }
 
+  /**
+    * Given an elastic search index, return the most recent document for that index.
+    *
+    * Document ordering is determined by the "upsert ID", which must be indexed field.
+    *
+    * @param index the elasticsearch index
+    * @tparam D the document type this index contains
+    * @return the most recent document for this index, if any
+    */
+  def getMostRecentDocument[D: HitReader](
```

Comment:
`D <: ClioDocument` will make sure that the type we're searching over actually contains an `upsert_id` field. It'd probably make sense to add the bound to the `ElasticsearchIndex` class too.

Reply:
OK